search/searcher/optimize_knn.go

```diff
@@ -0,0 +1,59 @@
+//  Copyright (c) 2023 Couchbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 		http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build vectors
+// +build vectors
+
+package searcher
+
+import (
+	"context"
+
+	"github.com/blevesearch/bleve/v2/search"
+	index "github.com/blevesearch/bleve_index_api"
+)
+
+func optimizeKNNSearcher(ctx context.Context, optimizationKind string,
+	indexReader index.IndexReader, qsearchers []search.Searcher,
+	options search.SearcherOptions) error {
+	var octx index.VectorOptimizableContext
+
+	for _, searcher := range qsearchers {
+		// Only applicable to KNN Searchers.
+		if _, ok := searcher.(*KNNSearcher); !ok {
```

Comment:
since the implementation check of VectorOptimizable is applicable only for knn searchers, i don't think this check is necessary - because the check at 38 passes only if its a knn searcher.